src/Razor/test/Microsoft.VisualStudio.Razor.Integration.Test/AbstractEditorTest.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Razor.Integration.Test.InProcess;
+
+namespace Microsoft.VisualStudio.Razor.Integration.Test
+{
+    public abstract class AbstractEditorTest : AbstractIntegrationTest
+    {
+        private readonly string? _solutionName;
+        private readonly string? _projectName;
+        private readonly string? _projectTemplate;
+
+        protected AbstractEditorTest()
+        {
+        }
+
+        protected AbstractEditorTest(string solutionName)
+            : this(solutionName, WellKnownProjectTemplates.BlazorProject, "BlazorProject")
+        {
+        }
+
+        protected AbstractEditorTest(string solutionName, string projectTemplate, string projectName)
+        {
+            _solutionName = solutionName;
+            _projectTemplate = projectTemplate;
+            _projectName = projectName;
+        }
+
+        protected abstract string LanguageName { get; }
+
+        public override async Task InitializeAsync()
+        {
+            await base.InitializeAsync();
+
+            if (_solutionName != null)
+            {
+                RoslynDebug.AssertNotNull(_projectTemplate);
+                RoslynDebug.AssertNotNull(_projectName);
+
+                await TestServices.SolutionExplorer.CreateSolutionAsync(_solutionName, HangMitigatingCancellationToken);
+                await TestServices.SolutionExplorer.AddProjectAsync(_projectName, _projectTemplate, LanguageName, HangMitigatingCancellationToken);
+                await TestServices.SolutionExplorer.RestoreNuGetPackagesAsync(ProjectName, HangMitigatingCancellationToken);
+
+                await TestServices.Editor.SetUseSuggestionModeAsync(false, HangMitigatingCancellationToken);
```

Comment:
CC @sharwell to correct me if I get this wrong since he wrote it (I accidentally wiped his commits in a rebase) and it seems to be a re-use of [what was done in Roslyn](https://github.com/dotnet/roslyn/blob/f948cdcb6956a6c0cfeecb45e0feccd62060d334/src/VisualStudio/IntegrationTest/New.IntegrationTests/AbstractEditorTest.cs#L55).

I'm believe this is done to prevent completions/suggestions from being offered unless explicitly requested, minimizing the amount of work done and preventing an "accidental completion" from confusing our expectations.

Agree that it's less then clear though, so I'll add a comment either way.

Reply:
Short answer is some tests in Roslyn change this value, and we don't want those tests to fail to restore the value to default.

Slightly longer answer is Roslyn now has `StateResetInProcess`, which should provide the complete strategy for returning the IDE to a known default state between each test, ensuring that test A does not impact the outcome of later test B regardless of whether A passes/fails.